### ISSUE TITLE
vertical_whitespace_between_cases rule is not handling 1 edge case properly

### ISSUE THREAD

**Issue by LeonidKokhnovich** [realm/SwiftLint#2760]
### New Issue Checklist

- [+] Updated SwiftLint to the latest version
- [+] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug
When vertical_whitespace_between_cases is enabled and I run "swiftlint autocorrect", the following code is not updated:
```swift
public enum ConnectionState: String {
    case idle
    case connecting

    init?(from string: String) {
        switch string {
        case "Idle":
            self = .idle
        case "Connecting":
            self = .connecting

        default:
            return nil
        }
    }
}
```
The expected code would be:
```swift
public enum ConnectionState: String {
    case idle
    case connecting

    init?(from string: String) {
        switch string {
        case "Idle":
            self = .idle

        case "Connecting":
            self = .connecting

        default:
            return nil
        }
    }
}
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint autocorrect
Loading configuration from '.swiftlint.yml'
Correcting Swift files at paths 
...
Correcting 'FortiExtensionDevice.swift' (26/368)
...
Done correcting 368 files!
```
FortiExtensionDevice.swift - file where the enum is declared.

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 
0.32.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
homebrew, pods
* Paste your configuration file:
```yml
opt_in_rules:
  - vertical_whitespace_between_cases
```

* Which Xcode version are you using (check `xcode-select -p`)?
Xcode 10.2
